duckplayer/duckplayer-impl/src/main/java/com/duckduckgo/duckplayer/impl/DuckPlayerLocalFilesPath.kt

```diff
@@ -17,51 +17,38 @@
 package com.duckduckgo.duckplayer.impl
 
 import android.content.res.AssetManager
-import com.duckduckgo.app.di.AppCoroutineScope
 import com.duckduckgo.common.utils.DispatcherProvider
 import com.duckduckgo.di.scopes.AppScope
 import com.squareup.anvil.annotations.ContributesBinding
 import dagger.SingleInstanceIn
-import java.io.IOException
 import javax.inject.Inject
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Deferred
-import kotlinx.coroutines.async
-import kotlinx.coroutines.withContext
+import kotlinx.coroutines.runBlocking
 
 interface DuckPlayerLocalFilesPath {
-    suspend fun assetsPath(): List<String>
+    fun assetsPath(): List<String>
 }
 
 @ContributesBinding(AppScope::class)
 @SingleInstanceIn(AppScope::class)
 class RealDuckPlayerLocalFilesPath @Inject constructor(
     private val assetManager: AssetManager,
-    @AppCoroutineScope appCoroutineScope: CoroutineScope,
-    private val dispatcherProvider: DispatcherProvider,
+    dispatcherProvider: DispatcherProvider,
 ) : DuckPlayerLocalFilesPath {
 
-    private val assetsPathDeferred: Deferred<List<String>> = appCoroutineScope.async(dispatcherProvider.io()) {
-        getAllAssetFilePaths("duckplayer")
-    }
+    private val assetsPaths: List<String> = runBlocking(dispatcherProvider.io()) { getAllAssetFilePaths("duckplayer") }
 
-    override suspend fun assetsPath(): List<String> {
-        return withContext(dispatcherProvider.io()) {
-            assetsPathDeferred.await()
-        }
-    }
+    override fun assetsPath(): List<String> = assetsPaths
 
     private fun getAllAssetFilePaths(directory: String): List<String> {
         val filePaths = mutableListOf<String>()
         val files = assetManager.list(directory) ?: return emptyList()
 
         files.forEach {
             val fullPath = "$directory/$it"
-            try {
```

Comment:
Why removing the try/catch? Calling `assetManager.list` can throw an IOException still. I assume is because we have full control over the assets here but FYI.

Reply:
We do, but I've added them back to be on the safe side, in case there's something wrong with the filesystem or whatever